Generate data stream input configuration and documentation

mjwolf opened the pull request:
This change introduces a new interactive prompt to the create data-stream command. When a user creates a data stream of type logs, they can now select from a list of common input types (e.g., aws-s3, gcp-pubsub, udp, tcp).

Based on the user's selection, the command now automatically populates the data_stream/<name>/manifest.yml with the corresponding input streams and their required variables. This streamlines the creation process for new log-based data streams by providing sensible defaults and reducing manual configuration.

This also updates the package README generation step to support a new function "inputDocs".  When this is used in the readme template, all inputs used in the package will be found, and all will be listed in the readme. For inputs which have additional documentation added to elastic-package, the input documentation will also be inserted into the rendered readme.

*Key changes include:*

Added a multi-select prompt to the create data-stream command for choosing log input types.

Added `internal/packages/_static/inputs/*.yml` files, which contains information on each input type supported by the create command, with information on default name, title, description and variables.

Introduced internal/packages/archetype/data_stream_inputs.go to manage the logic for populating input variables from the new static inputs yml files.

Updated the dataStream-manifest.yml.tmpl to dynamically render streams and their associated variables.

Refactored the packages.DataStreamManifest to use a new packages.Stream struct for better organization.

Add "inputDocs" to the readme render functions, which will find all inputs used in the package, and render documentation for them.

Add `docs/_static/inputs/*.yml` for keeping input documentation (this is separate from archetype yml file, to avoid a circular dependancy between the packages). 

*How to test:*

**Create command**
Run the elastic-package create data-stream command.

Select logs as the data stream type.

Select one or more input types from the list.

Verify that the generated data_stream/<name>/manifest.yml contains the correct streams configuration for the inputs you selected.

**Build readme**
Update a package readme to have the {{inputDocs}} function in the readme template

Build the package

Check the generated readme for the input documentation.

mjwolf commented on cmd/create_data_stream.go:
```diff
@@ -81,7 +82,7 @@ func createDataStreamCommandAction(cmd *cobra.Command, args []string) error {
 			Prompt: &survey.Confirm{
 				Message: "Enable creation of subobjects for fields with dots in their names?",
-				Default: false,
+				Default: true,
```
@jsoriano I changed this for two reasons.
1. True is the default value in the [package-spec](https://github.com/elastic/package-spec/blob/561d1bc653724d549c527fba82f64f6b25d8cff1/spec/integration/data_stream/manifest.spec.yml#L479)
2. When set to false, the created manifest can't be installed, and I wanted the default options to create a working package.  I get an error like this when trying to install a package with a datastream with this option set to false.
```
Error: package installation failed: failed to build package: invalid content found in built zip package: found 1 validation error:
   1. file "/home/mwolf/git/integrations/build/packages/endace-0.3.0.zip/data_stream/stream2/manifest.yml" is invalid: field elasticsearch.index_template.mappings: Additional property subobjects is not allowed```

This isn't something introduced here, it also happens with elastic-package 0.114.0. I don't know enough about what this option should do to fix it properly

kgeller reviewed (commented):
Some general feedback/questions: 

1) It feels a bit awkward without the changes to the agent/stream template since the manifest will align with what's there. I know we have that slotted for a subtask, but I feel it would make more sense to include all together. Unless we can guarantee they'll be merged together around the same time and will definitely make it into the same release

2) I appreciate that the user doesn't have to select an input, but then we get a pretty empty generated manifest file (and later no agent/stream files). I feel it could be beneficial if no input is selected, we fall back to what gets generated today with the currently existing logfile details 

3) Not sure how possible this is, but I think it could also be great if the manifest sections could be broken out into their own files. The one big massive file feels less ideal for maintaining

4) Is there anything we can do sync up the package manifest with the new datastream details?

mjwolf commented on cmd/create_data_stream.go:
```diff
@@ -126,6 +127,41 @@ func createDataStreamCommandAction(cmd *cobra.Command, args []string) error {
+				Prompt: &survey.MultiSelect{
+					Message: "Select input types which will be used in this data stream",
+					Options: []string{
+						"aws-cloudwatch",
```
I thought about that too. It would be nice, but I don't think it's possible with the survey module we're using.

It might be nice to switch to another TUI module which can do more in the future, but that'll be a big job.  For now I can add a link to the input documentation in the first prompt.

I think that's the best that can be done with this module.

mjwolf commented:
> Some general feedback/questions:
> 
> 1. It feels a bit awkward without the changes to the agent/stream template since the manifest will align with what's there. I know we have that slotted for a subtask, but I feel it would make more sense to include all together. Unless we can guarantee they'll be merged together around the same time and will definitely make it into the same release

I agree with that now. I was thinking that it'll be a big PR to have both changes in the same PR. But now that I'm working on the 2nd, a lot of the changes are closely related, and I think it makes sense to do them both at the same time.

So I converted this PR to draft, and I'll keep working on it to do them both at the same time.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/elastic-package/builds/5929)
* Commit: f72eeeb49bfad2b14a79aababbdc3f4c737596d0

### Failed CI Steps
* [:linux: Run unit tests](https://buildkite.com/elastic/elastic-package/builds/5929#0198afc2-6875-4203-86da-011a976b4d2c)

### History
* :broken_heart: [Build #5928](https://buildkite.com/elastic/elastic-package/builds/5928) failed 69c6854c55cc53bad24a963625aa7abfc8da1703
* :green_heart: [Build #5925](https://buildkite.com/elastic/elastic-package/builds/5925) succeeded 498cee7ab15cd3d2949b2176d999612f80d9124b
* :broken_heart: [Build #5924](https://buildkite.com/elastic/elastic-package/builds/5924) failed 85f2e4c78425d26ba32810982d085d937a68cda4

<!--buildkite-pr-comment-elastic-package
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5929","number":5929,"commit":"f72eeeb49bfad2b14a79aababbdc3f4c737596d0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5928","number":5928,"commit":"69c6854c55cc53bad24a963625aa7abfc8da1703"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5925","number":5925,"commit":"498cee7ab15cd3d2949b2176d999612f80d9124b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5924","number":5924,"commit":"85f2e4c78425d26ba32810982d085d937a68cda4"}],"number":5929}
buildkite-pr-comment-->

kgeller commented on cmd/create_data_stream.go:
```diff
@@ -126,6 +127,41 @@ func createDataStreamCommandAction(cmd *cobra.Command, args []string) error {
+						"gcp-pubsub",
+						"gcs",
+						"http_endpoint",
+						"httpjson",
```
I still see it when testing locally and in the PR?

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/elastic-package/builds/5953)
* Commit: f411f9b8e886feb35a4a8d7ec82e7736864425a1

### Failed CI Steps
* [:go: Run check-static](https://buildkite.com/elastic/elastic-package/builds/5953#0198cefd-150a-4d9a-8d2a-db1348d46071)

### History
* :green_heart: [Build #5944](https://buildkite.com/elastic/elastic-package/builds/5944) succeeded eb5a25168580e98f5d3e7878087d99a2ffe72efc
* :broken_heart: [Build #5942](https://buildkite.com/elastic/elastic-package/builds/5942) failed 4d0e01e129f6ee7336d13196506ea2fe044af9f7
* :broken_heart: [Build #5941](https://buildkite.com/elastic/elastic-package/builds/5941) failed 4deb7c4959af041c0f55191d613cc9c9ed107ea7
* :broken_heart: [Build #5940](https://buildkite.com/elastic/elastic-package/builds/5940) failed 181d6e7270ad6273e16f422b88e150199795be02
* :broken_heart: [Build #5929](https://buildkite.com/elastic/elastic-package/builds/5929) failed f72eeeb49bfad2b14a79aababbdc3f4c737596d0
* :broken_heart: [Build #5928](https://buildkite.com/elastic/elastic-package/builds/5928) failed 69c6854c55cc53bad24a963625aa7abfc8da1703

<!--buildkite-pr-comment-elastic-package
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5953","number":5953,"commit":"f411f9b8e886feb35a4a8d7ec82e7736864425a1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5944","number":5944,"commit":"eb5a25168580e98f5d3e7878087d99a2ffe72efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5942","number":5942,"commit":"4d0e01e129f6ee7336d13196506ea2fe044af9f7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5941","number":5941,"commit":"4deb7c4959af041c0f55191d613cc9c9ed107ea7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5940","number":5940,"commit":"181d6e7270ad6273e16f422b88e150199795be02"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5929","number":5929,"commit":"f72eeeb49bfad2b14a79aababbdc3f4c737596d0"}],"number":5953}
buildkite-pr-comment-->

mjwolf commented:
> > This is what the `panw` integration rendered README looks like in Kibana when using this markdown in the "References" section:
> 
> @mjwolf is it possible to take advantage of the collapsable sections here with md now that we have support for that in kibana? I worry these big sections at the bottom will create unnecessary noise for the users when browsing the page


I've changed to use collapsible sections for each input's doc (and updated the screenshot above)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/elastic-package/builds/5954)
* Commit: d320f9e81d3c82e2f465a0a8d6acf3042ba226be

### Failed CI Steps
* [:go: Run check-static](https://buildkite.com/elastic/elastic-package/builds/5954#0198cf02-afcf-4687-a4e2-a1a249332e03)

### History
* :broken_heart: [Build #5953](https://buildkite.com/elastic/elastic-package/builds/5953) failed f411f9b8e886feb35a4a8d7ec82e7736864425a1
* :green_heart: [Build #5944](https://buildkite.com/elastic/elastic-package/builds/5944) succeeded eb5a25168580e98f5d3e7878087d99a2ffe72efc
* :broken_heart: [Build #5942](https://buildkite.com/elastic/elastic-package/builds/5942) failed 4d0e01e129f6ee7336d13196506ea2fe044af9f7
* :broken_heart: [Build #5941](https://buildkite.com/elastic/elastic-package/builds/5941) failed 4deb7c4959af041c0f55191d613cc9c9ed107ea7
* :broken_heart: [Build #5940](https://buildkite.com/elastic/elastic-package/builds/5940) failed 181d6e7270ad6273e16f422b88e150199795be02

<!--buildkite-pr-comment-elastic-package
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5954","number":5954,"commit":"d320f9e81d3c82e2f465a0a8d6acf3042ba226be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5953","number":5953,"commit":"f411f9b8e886feb35a4a8d7ec82e7736864425a1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5944","number":5944,"commit":"eb5a25168580e98f5d3e7878087d99a2ffe72efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5942","number":5942,"commit":"4d0e01e129f6ee7336d13196506ea2fe044af9f7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5941","number":5941,"commit":"4deb7c4959af041c0f55191d613cc9c9ed107ea7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5940","number":5940,"commit":"181d6e7270ad6273e16f422b88e150199795be02"}],"number":5954}
buildkite-pr-comment-->

mjwolf commented on cmd/create_data_stream.go:
```diff
@@ -126,6 +127,42 @@ func createDataStreamCommandAction(cmd *cobra.Command, args []string) error {
+						"redis",
+						"tcp",
+						"udp",
+						"winlog",
```
I don't know of a source of truth that could be used for this. The inputs here were selected because they are the most-used. I think the idea of composable inputs could be good to become the source of truth.  It would be good if that had some concept of deprecated/discouraged inputs, since there's now some inputs that we don't want to add new usages.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/elastic-package/builds/5955)
* Commit: 5c747e8f8814d4b469dd00ca7015820624abd935

### History
* :broken_heart: [Build #5954](https://buildkite.com/elastic/elastic-package/builds/5954) failed d320f9e81d3c82e2f465a0a8d6acf3042ba226be
* :broken_heart: [Build #5953](https://buildkite.com/elastic/elastic-package/builds/5953) failed f411f9b8e886feb35a4a8d7ec82e7736864425a1
* :green_heart: [Build #5944](https://buildkite.com/elastic/elastic-package/builds/5944) succeeded eb5a25168580e98f5d3e7878087d99a2ffe72efc
* :broken_heart: [Build #5942](https://buildkite.com/elastic/elastic-package/builds/5942) failed 4d0e01e129f6ee7336d13196506ea2fe044af9f7
* :broken_heart: [Build #5941](https://buildkite.com/elastic/elastic-package/builds/5941) failed 4deb7c4959af041c0f55191d613cc9c9ed107ea7

<!--buildkite-pr-comment-elastic-package
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5955","number":5955,"commit":"5c747e8f8814d4b469dd00ca7015820624abd935"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5954","number":5954,"commit":"d320f9e81d3c82e2f465a0a8d6acf3042ba226be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5953","number":5953,"commit":"f411f9b8e886feb35a4a8d7ec82e7736864425a1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5944","number":5944,"commit":"eb5a25168580e98f5d3e7878087d99a2ffe72efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5942","number":5942,"commit":"4d0e01e129f6ee7336d13196506ea2fe044af9f7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5941","number":5941,"commit":"4deb7c4959af041c0f55191d613cc9c9ed107ea7"}],"number":5955}
buildkite-pr-comment-->

jsoriano commented on cmd/create_data_stream.go:
```diff
@@ -126,6 +127,80 @@ func createDataStreamCommandAction(cmd *cobra.Command, args []string) error {
+					},
+					PageSize: 50,
+					Description: func(value string, index int) string {
+						switch value {
+						case "aws-cloudwatch":
+							return "AWS Cloudwatch"
+						case "aws-s3":
+							return "AWS S3"
+						case "azure-blob-storage":
+							return "Azure Blob Storage"
+						case "azure-eventhub":
+							return "Azure Eventhub"
+						case "cel":
+							return "Common Expression Language (CEL)"
+						case "entity-analytics":
+							return "Entity Analytics"
+						case "etw":
+							return "Event Tracing for Windows (ETW)"
+						case "filestream":
+							return "Filestream"
+						case "gcp-pubsub":
+							return "GCP PubSub"
+						case "gcs":
+							return "Google Cloud Storage (GCS)"
+						case "http_endpoint":
+							return "HTTP Endpoint"
+						case "journald":
+							return "Journald"
+						case "netflow":
+							return "Netflow"
+						case "redis":
+							return "Redis"
+						case "tcp":
+							return "TCP"
+						case "udp":
+							return "UDP"
+						case "winlog":
+							return "WinLogBeat"
+						}
```
To ensure that we keep this list updated with the options of the multiselect, we could keep this info in a map, and use the ordered keys as options for the multiselect.
```suggestion
						return inputOptions[value]
```

mjwolf replied:
Thanks, it's a good idea to use a map. I've done that, so there's now just one map for the names and descriptions, instead of the duplicated lists.

jsoriano commented on internal/docs/resources.go:
```diff
@@ -0,0 +1,84 @@
+	inputTcp,
+	inputUdp,
+	inputWinlog,
+}
```
This list is duplicated in `internal/packages/archetype/resources.go`? Could we reuse it?

jsoriano commented on internal/docs/resources.go:
```diff
@@ -0,0 +1,84 @@
+
+// Input definitions
+
+//go:embed _static/inputs/aws-cloudwatch.yml
```
Is it also possible to embed a whole directory. In case this helps, maybe when adding new inputs in the future.
```
//go:embed _static/inputs
var inputResources embed.FS
```

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/elastic-package/builds/5959)
* Commit: fc75addc07b8ea65093bf955b622606e86a1bc4d

### History
* :green_heart: [Build #5955](https://buildkite.com/elastic/elastic-package/builds/5955) succeeded 5c747e8f8814d4b469dd00ca7015820624abd935
* :broken_heart: [Build #5954](https://buildkite.com/elastic/elastic-package/builds/5954) failed d320f9e81d3c82e2f465a0a8d6acf3042ba226be
* :broken_heart: [Build #5953](https://buildkite.com/elastic/elastic-package/builds/5953) failed f411f9b8e886feb35a4a8d7ec82e7736864425a1
* :green_heart: [Build #5944](https://buildkite.com/elastic/elastic-package/builds/5944) succeeded eb5a25168580e98f5d3e7878087d99a2ffe72efc
* :broken_heart: [Build #5942](https://buildkite.com/elastic/elastic-package/builds/5942) failed 4d0e01e129f6ee7336d13196506ea2fe044af9f7
* :broken_heart: [Build #5941](https://buildkite.com/elastic/elastic-package/builds/5941) failed 4deb7c4959af041c0f55191d613cc9c9ed107ea7

<!--buildkite-pr-comment-elastic-package
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5959","number":5959,"commit":"fc75addc07b8ea65093bf955b622606e86a1bc4d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5955","number":5955,"commit":"5c747e8f8814d4b469dd00ca7015820624abd935"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5954","number":5954,"commit":"d320f9e81d3c82e2f465a0a8d6acf3042ba226be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5953","number":5953,"commit":"f411f9b8e886feb35a4a8d7ec82e7736864425a1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5944","number":5944,"commit":"eb5a25168580e98f5d3e7878087d99a2ffe72efc"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5942","number":5942,"commit":"4d0e01e129f6ee7336d13196506ea2fe044af9f7"}],"number":5959}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/elastic-package/builds/5962)
* Commit: c66ca35825994844010609f20fa85843ef16dc40

### Failed CI Steps
* [:go: Run check-static](https://buildkite.com/elastic/elastic-package/builds/5962#0198df89-265f-453a-80c1-c1fd6d16bd4e)

### History
* :green_heart: [Build #5959](https://buildkite.com/elastic/elastic-package/builds/5959) succeeded fc75addc07b8ea65093bf955b622606e86a1bc4d
* :green_heart: [Build #5955](https://buildkite.com/elastic/elastic-package/builds/5955) succeeded 5c747e8f8814d4b469dd00ca7015820624abd935
* :broken_heart: [Build #5954](https://buildkite.com/elastic/elastic-package/builds/5954) failed d320f9e81d3c82e2f465a0a8d6acf3042ba226be
* :broken_heart: [Build #5953](https://buildkite.com/elastic/elastic-package/builds/5953) failed f411f9b8e886feb35a4a8d7ec82e7736864425a1
* :green_heart: [Build #5944](https://buildkite.com/elastic/elastic-package/builds/5944) succeeded eb5a25168580e98f5d3e7878087d99a2ffe72efc
* :broken_heart: [Build #5942](https://buildkite.com/elastic/elastic-package/builds/5942) failed 4d0e01e129f6ee7336d13196506ea2fe044af9f7

<!--buildkite-pr-comment-elastic-package
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5962","number":5962,"commit":"c66ca35825994844010609f20fa85843ef16dc40"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5959","number":5959,"commit":"fc75addc07b8ea65093bf955b622606e86a1bc4d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5955","number":5955,"commit":"5c747e8f8814d4b469dd00ca7015820624abd935"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5954","number":5954,"commit":"d320f9e81d3c82e2f465a0a8d6acf3042ba226be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5953","number":5953,"commit":"f411f9b8e886feb35a4a8d7ec82e7736864425a1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/elastic-package/builds/5944","number":5944,"commit":"eb5a25168580e98f5d3e7878087d99a2ffe72efc"}],"number":5962}
buildkite-pr-comment-->

mjwolf commented:
> 4. Is there anything we can do sync up the package manifest with the new datastream details?

@kgeller, I had updated this to add a new policy_template to the package manifest when a new data stream is created, but it then caused some unit tests to fail. I think there's some interaction with the package-spec that makes it a bit more complicated then I thought. What do you think of leaving this out of this PR, since it's already pretty big.

We could look at updating the package spec later.

(This is what I did to update the package manifest when a data stream is added: https://github.com/elastic/elastic-package/pull/2826/commits/05acf8772b849e2543ff6ebf936604fc41aa7ea4)

mjwolf commented on internal/docs/_static/inputs/cel.yml:
```diff
@@ -0,0 +1,397 @@
+    description: Client secret used for OAuth2 authentication
+    show_user: true
+    required: false
+    secret: true
```
The Secret field wasn't in the updated manifest template, so it wasn't added into the resulting manifest, even though it was in the inputs yaml definition.

I've added secrets, and other missing fields to the manifest template now.
